x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/TransportMountSearchableSnapshotAction.java

```diff
@@ -250,6 +254,8 @@ protected void masterOperation(
                 dataTierAllocationSetting.get(indexSettings);
             }
 
+            clusterInfoService.syncClusterInfo();
```

Comment:
I think this do not fixes the issue, at this point the index is not mounted yet. I think this pass the test as the implementation in `InternalClusterInfoService#syncClusterInfo` triggers an async operation and at the point it refreshes the information the index is already mounted I think?